ui/pages/send/send-footer/send-footer.component.js

```diff
@@ -100,35 +59,10 @@ export default class SendFooter extends Component {
     });
   }
 
-  formShouldBeDisabled() {
-    const {
-      data,
-      inError,
-      sendToken,
-      tokenBalance,
-      gasTotal,
-      to,
-      gasLimit,
-      gasIsLoading,
-      noGasPrice,
-    } = this.props;
-    const missingTokenBalance = sendToken && !tokenBalance;
-    const gasLimitTooLow = gasLimit < 5208; // 5208 is hex value of 21000, minimum gas limit
-    const shouldBeDisabled =
-      inError ||
-      !gasTotal ||
-      missingTokenBalance ||
-      !(data || to) ||
-      gasLimitTooLow ||
-      gasIsLoading ||
-      noGasPrice;
-    return shouldBeDisabled;
-  }
-
   componentDidUpdate(prevProps) {
-    const { inError, sendErrors } = this.props;
+    const { disabled, sendErrors } = this.props;
     const { metricsEvent } = this.context;
-    if (!prevProps.inError && inError) {
+    if (!prevProps.disabled && disabled && Object.keys(sendErrors).length > 0) {
```

Comment:
I think this should be checking just for whether errors were introduced, irrespective of whether the button is disabled. Otherwise, if an error is introduced while the button is disabled for another reason, the metric event will get missed.

e.g. for example if an error shows up before gas has finished loading, it'll be skipped here.

Reply:
https://github.com/MetaMask/metamask-extension/pull/10965/commits/ace191f212b7031c777fe00f96cf68d352d976d4